src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -147,6 +165,149 @@ public void ConfigurationOrderingTest(ConfigurationLevel level)
+            builder.ReplaceAspnetEnvironment(null);
+            builder.ReplaceDotnetEnvironment(null);
+            builder.ReplaceMonitorEnvironment(null);
+
+            // Build the host and get the Urls property from configuration.
+            IHost host = builder.Build();
+            IConfiguration rootConfiguration = host.Services.GetRequiredService<IConfiguration>();
+
+            Stream stream = new MemoryStream();
+
+            using ConfigurationJsonWriter jsonWriter = new ConfigurationJsonWriter(stream);
+            jsonWriter.Write(rootConfiguration, full: !redact, skipNotPresent: false);
+            jsonWriter.Dispose();
+
+            stream.Position = 0;
+
+            using (var streamReader = new StreamReader(stream))
+            {
+                string configString = streamReader.ReadToEnd();
+
+                _outputHelper.WriteLine(configString);
+
+                CompareOutput(configString, expectedConfiguration);
+            }
+        }
+
+        private string ConstructUserSettingsJson()
+        {
+            string generatedUserSettings = "{";
+
+            string[] fileNames = Directory.GetFiles(Path.Combine(Directory.GetCurrentDirectory(), "SampleConfigurations"));
+
+            foreach (var fileName in fileNames)
+            {
+                string fileContents = File.ReadAllText(fileName);
+
+                fileContents = fileContents.Remove(fileContents.LastIndexOf("}"));
+                fileContents = fileContents.Remove(fileContents.IndexOf("{"), 1);
+
+                generatedUserSettings += fileContents + (!fileName.Equals(fileNames.Last()) ? "," : "");
+            }
+
+            generatedUserSettings += "}";
+
+            return generatedUserSettings;
+        }
+
+        private void CompareOutput(string actual, string expected)
+        {
+            Assert.Equal(CleanWhitespace(expected), CleanWhitespace(actual));
+        }
+
+        private string ConstructExpectedOutput(bool redact)
+        {
+            Dictionary<string, string> categoryMapping = GetConfigurationFileNames(redact);
+
+            string expectedOutput = "{";
+
+            foreach (var key in orderedConfigurationKeys)
+            {
+                expectedOutput += "\"" + key + "\"" + ":";
+
+                if (categoryMapping.ContainsKey(key))
+                {
+                    string expectedPath = Path.Combine(Directory.GetCurrentDirectory(), "ExpectedConfigurations", categoryMapping[key]);
+                    expectedOutput += File.ReadAllText(expectedPath);
+                }
+                else
+                {
+                    expectedOutput += "\"" + Strings.Placeholder_NotPresent + "\"";
+                }
+
+                expectedOutput += (!key.Equals(orderedConfigurationKeys.Last())) ? "," : "";
+            }
+
+            expectedOutput += "}";
+
+            return expectedOutput;
+        }
+
+        private string CleanWhitespace(string rawText)
```

Comment:
This may not be necessary if you are using System.Text.Json to serialize and write. They have options for setting the indent to empty string and typically minify their outputs by default: https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-6-0#how-to-write-net-objects-as-json-serialize